### PR TITLE
ET-3440 Adding basic postgres files

### PR DESCRIPTION
**Pull request by jorge-cr-13**
## What 🕵️ 🔍

- Adding a Postgres DB for our cluster

## References 📝 🔗

- [Jira task](https://xainag.atlassian.net/browse/)

## Reviewer checklist

*Reviewer agreement:*

* Reviewers shouldn't review a "Draft Pull Request", but rather wait for it to be marked as "Ready for Review".
* Reviewers do **not** commit or merge the pull request.
* If/when there are changes that need to be applied (applying Terraform or deploying Kubernetes manifests), the **assignee** is the one responsible for doing it.
* Use the labels `not-applied`, `partially-applied` and `fully-applied` so others know the current progress.
  * Remember to change labels accordingly once you start and finish to apply the changes.

### CODE REVIEW

File: aws/services/postgres/main.tf [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  name                   = "postgres-db"
+  instance_class         = "db.t2.micro"
+  allocated_storage      = 5
+  engine                 = "postgres"
+  engine_version         = "12.5"
+  skip_final_snapshot    = true
+  publicly_accessible    = true
+  vpc_security_group_ids = [aws_security_group.db_security_group.id]
+  username               = "postgres"
+  password               = "random_string.db-password.result}"
```
Comment (Robert-Steiner) [inline aws/services/postgres/main.tf:29]: here we are missing the field [multi_az](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/db_instance#multi_az)

File: aws/services/postgres/main.tf [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    from_port   = 5432
+    to_port     = 5432
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+}
+resource "aws_db_instance" "postgres_db_instance" {
+  identifier             = "postgres-db"
+  name                   = "postgres-db"
+  instance_class         = "db.t2.micro"
```
Comment (Robert-Steiner) [inline aws/services/postgres/main.tf:21]: just noticed that `db.t4g.micro` is even cheaper 
```suggestion
  instance_class         = "db.t4g.micro"
```

**Review by Robert-Steiner — COMMENTED** [review]
I'm still thinking about using aurora instead of plain rds. aurora has some nice features that could save us some time. for dev we could use serverless v2. wdyt?

**Comment by jorge-cr-13** [comment]
@Robert-Steiner sorry I opened this PR, but I'm still working on it and I forgot to put it on draft.

**Comment by jorge-cr-13** [comment]
> I'm still thinking about using aurora instead of plain rds. aurora has some nice features that could save us some time. for dev we could use serverless v2. wdyt?

I think it would be quite cool to use it. I think for now we should go with Postgres and then sure why not to change. :)

File: aws/stage/rds/outputs.tf [hunk 3]
```diff
@@ -0,0 +1,3 @@
+output "db_endpoint" {
+  value = aws_db_instance.postgres.endpoint
```
Comment (Robert-Steiner) [inline aws/stage/rds/outputs.tf:2]: with address we have the endpoint + port
```suggestion
  value = aws_db_instance.postgres.address
```

File: aws/stage/rds/main.tf [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  instance_class                  = "db.t4g.micro"
+  allocated_storage               = var.allocated_storage
+  max_allocated_storage           = var.max_allocated_storage
+  engine                          = "postgres"
+  skip_final_snapshot             = true
+  publicly_accessible             = false
+  vpc_security_group_ids          = [aws_security_group.postgres.id]
+  username                        = var.username
+  password                        = "random_string.db-password.result"
+  availability_zone               = data.aws_availability_zones.available_zones.names[0]
```
Comment (Robert-Steiner) [inline aws/stage/rds/main.tf:50]: I think we can remove it and let aws decide on which az to deploy the primary node
```suggestion
```

File: aws/stage/rds/main.tf [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+data "aws_availability_zones" "available_zones" {
+  state = "available"
+}
+
+resource "random_string" "db-password" {
+  length  = 32
+  upper   = true
+  numeric = true
+  special = false
+}
```
Comment (Robert-Steiner) [inline aws/stage/rds/main.tf:19]: for the `db_name`, `username` and `password` i would like to use the parameter store.
we can create them manually via the console ui and then use them via a `data source` (it is similar what we do with the remote state)

furthermore we can use the above parameters + the db address and construct the postgres url and define it as a new parameter. this way we don't need to adjust the url after we changed for example the username

```hcl
data "aws_ssm_parameter" "postgres_admin_password" {
  name = "postgres_admin_password"
}

data "aws_ssm_parameter" "postgres_admin_username" {
  name = "postgres_admin_username"
}

data "aws_ssm_parameter" "postgres_db_name" {
  name = "postgres_db_name"
}

resource "aws_ssm_parameter" "postgres_url" {
  name        = "postgres_url"
  description = "Connection string of the postgres cluster"
  type        = "SecureString"
  value       = "postgresql://${data.aws_ssm_parameter.postgres_admin_username}:${data.aws_ssm_parameter.postgres_admin_password}@${aws_db_instance.postgres.address}/${data.aws_ssm_parameter.postgres_db_name}"
}
```

username can be `user`, for the pw you can use 1password to create one and the db_name can be `xayn`

**Review by Robert-Steiner — APPROVED** [review]
nice! 🚀

**Pull request merged**